onnxruntime/core/session/inference_session.cc

```diff
@@ -253,6 +254,36 @@ void InferenceSession::ConstructorCommon(const SessionOptions& session_options,
   // after the invocation of FinalizeSessionOptions.
   InitLogger(logging_manager_);  // this sets session_logger_ so that it can be used for logging after this point.
 
+  LOGS(*session_logger_, INFO) << "Session Options { "
```

Comment:
Might be better to either define an ```operator<<``` in session_options.h/cc or add a ToString() method in SessionOptions; each object knows how to print/stringify itself and the options stay close to the printing code. Then we can simply say ```LOGS(*session_logger_, INFO) << "Session Options " << session_options_;``` or ```<< session_options_.ToString();```

Reply:
Per your suggestion I did move implementation closer to the definition.